instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -44,6 +44,18 @@ def query(sql, options = {})
 
           private
 
+          def obfuscate_sql(sql)
+            # if ENV variable is set, removes any string values or numbers inside spaces
+            # (to avoid mathcing dates).
+            return sql unless ENV['OTEL_INSTRUMENTATION_MYSQL2_OBFUSCATE']
+
+            if sql.size > 2000
+              'SQL query too large to remove sensitive data ...'
+            else
+              sql.gsub(/'.+?'/m, '?').gsub(/ \d+ /m, ' ? ')
+            end
```

Comment:
It might be preferable to adopt the NewRelic obfuscation rules: https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/database/obfuscation_helpers.rb

Reply:
Do you mean to use their mysql related rules or bring the file as a shared tool that other instrumentations may use in the future?
Is it proper to copy from that codebase? I do not know if there will be licensing issues or something.

Reply:
I was proposing to use the mysql related rules. We can generalize later when we add other instrumentation.

That code is under an Apache license, so a) it can be copied, and b) it is compatible with our license.